### DISCUSSION TITLE
I setting variable of class name of vue how can I get the class name after vue render by `tailwindcss CLI`?

### DISCUSSION THREAD

**Discussion by carveybunt** [tailwindlabs/tailwindcss#4775] [Help]
For example, my Vue component:
```js
<template>
  <p v-if="$slots.default || !!description" :class="[..., descriptionSize ? 'text-' + descriptionSize : 'text-xs']">
    <slot>
      {{ description }}
    </slot>
  </p>
</template>
<script lang='ts'>
     export default defineComponent({
         ...
       props: {
          descriptionSize:{
              type:String,
              default:'',
           },
       },
          ...
</script>
```
I wanna use `tailwindcss CLI` to make `tailwind.css` after I running `npm run dev`. I set to value `lg` for `descriptionSize` when I used my Vue component, but `tailwind.css` doesn't have a class name for `text-lg`. What should I do to achieve that, someone can help me, please?

**Comment by dance2die** [comment]
_Note: Never used Vue_

If you don't use a full TW ("Tailwind") class name, then TW won't generate utilities/components for it.
because TW won't scan your file and interpolate class names.

```js
descriptionSize ? 'text-' + descriptionSize : 'text-xs'
```

won't work while following will.

```js
descriptionSize ? 'text-lg' : 'text-xs'
```

Since `descriptionSize` can be values other than `lg`, you can create a mapping object.

```js
const fontSizeMap = {
  'lg': 'text-lg',
  'xl': 'text-xl',
  '2xl': 'text-2xl'
}

// Nullish coalescing operator "??"
// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
fontSizeMap[descriptionSize] ?? 'text-xs';
```

TW will scan for those  font sizes and generate classes.

**Reply by carveybunt** [reply]
Thanks a lot, I got it.

**Reply by dance2die** [reply]
You're welcome there.
